ExampleMod/Content/NPCs/FallenSoul.cs

```diff
@@ -0,0 +1,137 @@
+using Microsoft.Xna.Framework;
+using System;
+using Terraria;
+using Terraria.ID;
+
+namespace ExampleMod.Content.NPCs
+{
+	// This NPC inherits from the HoverNPC abstract class included in ExampleMod, which is a customizable AI similar to that of Vanilla's Hover AI.
+	// The CustomBehavior and ShouldMove virtual methods are overridden here, as well as the acceleration and accelerationY field being set in the class constructor.
+	public class FallenSoul : HoverNPC
+	{
+		public FallenSoul() {
+			accelerationX = 0.06f;
+			accelerationY = 0.025f;
+		}
+
+		public override void SetStaticDefaults() => Main.npcFrameCount[npc.type] = Main.npcFrameCount[NPCID.Wraith];
+
+		public override void SetDefaults() {
+			npc.CloneDefaults(NPCID.Wraith); // Clone the default stats of a Wraith.
+			// Here we adjust the stats to our liking.
+			npc.width = 28;
+			npc.height = 36;
+			npc.aiStyle = -1;
+			npc.damage = 0;
+			npc.friendly = true;
+			npc.dontTakeDamageFromHostiles = true;
+			animationType = NPCID.Wraith;
+		}
+
+		// Allows hitting the NPC with items, even if it's friendly.
+		public override bool? CanBeHitByItem(Player player, Item item) => true;
+
+		// Allows hitting the NPC with friendly projectiles.
+		public override bool? CanBeHitByProjectile(Projectile projectile) => projectile.friendly && projectile.owner < 255;
+
+		// Allows us to do something when the NPC is hit. Here, we spawn dust.
+		public override void HitEffect(int hitDirection, double damage) {
+			if (npc.life > 0) {
+				for (int i = 0; i < damage / npc.lifeMax * 100; i++) {
+					Dust dust = Dust.NewDustDirect(npc.position, npc.width, npc.height, 192, hitDirection, -1f, 100, new Color(100, 100, 100, 100), 1f);
+					dust.noGravity = true;
+				}
+
+				return;
+			}
```

Comment:
needs more comments. I'd use an `if/else` here, but also, the only difference betewen the two loops is the number of dust, and the SpeedY, so you'd be better to merge them with some local variables declared at the top.